site/app/models/DockerUI.php

```diff
@@ -0,0 +1,501 @@
+    /** Set after parsing logs if this machine could not be updated */
+    public bool $failed_to_update;
+    /** Docker version on the machine */
+    public string $docker_version;
+    /** OS of the machine */
+    public string $os;
+    /** System information parsed from sysinfo logs */
+    public ?WorkerSystemInformation $system_information;
+    /** Create a new Worker object */
+    public function __construct(string $name, int $num_autograding_workers, array $capabilities, bool $is_enabled) {
+        $this->name = $name;
+        $this->num_autograding_workers = $num_autograding_workers;
+        $this->capabilities = $capabilities;
+        $this->is_enabled = $is_enabled;
+        //these pieces of information are known later in the parsing
+        $this->failed_to_update = false;
+        $this->system_information = null;
+        // this information is taken from the docker logs parsed earlier and associated after the worker obj is created
+        $this->docker_version = "Unknown";
+        $this->os = "Unknown";
+    }
+}
+
+/**
+ * Store information about a worker system information, this comes from system log information 
+ * instead of configuration
+ */
+class WorkerSystemInformation {
+    /** Name of worker machine this information is associated with */
+    public string $associated_worker; 
+    /** State of worker_service */
+    public string $worker_service;
+    /** State of worker_service */
+    public ?string $shipper_service;
+    /** State of worker_service */
+    public ?string $daemon_service;
+    /** Disk usage in percetange */
+    public string $disk_usage;
+    /** String with a unix load average
+     * See https://www.gnu.org/software/coreutils/manual/html_node/uptime-invocation.html#uptime-invocation 
+     * For understanding the load numbers
+     */
+    public string $load;
+    /** operating system of the worker */
+    public string $os;
+
+    /** Create a new WorkerSystemInformation object */
+    public function __construct(string $associated_worker, string $worker_service, string $shipper_service, string $daemon_service, 
+        string $disk_usage, string $load) {
+
+        $this->associated_worker = $associated_worker;
+        $this->worker_service = $worker_service;
+        $this->shipper_service = $shipper_service;
+        $this->daemon_service = $daemon_service;
+        $this->disk_usage = $disk_usage;
+        $this->load = $load;
+    }
+
+    /** 
+     * Given an associative array of worker system information and the worker name, convert it to a class and 
+     * perform null checks over the data
+     */
+    public static function fromArray(array $data, string $name): self {
+        return new self(
+            $name, 
+            $data['worker'] ?? 'No status collected',
+            $data['shipper'] ?? null,
+            $data['daemon'] ?? null,
+            $data['disk'] ?? 'No status collected',
+            $data['load'] ?? 'No status collected'
```

Comment:
Instead of setting an error message here, what do you think about setting missing data to null and then presenting the error text in the UI